### PR TITLE
Add GQL next stakes

### LINKED ISSUE #558: Problem: there is no way to query for next stakes via GraphQL
Add the next stakes resolver to support the call the `mina-block-explorer` uses for the frontend.

https://github.com/Granola-Team/mina-block-explorer/blob/main/graphql/queries/next_staking_ledgers.graphql

### PR DESCRIPTION
**Pull request by Isaac-DeFrain**
Resolves #558

### CODE REVIEW

**Comment by Isaac-DeFrain** [comment]
Rebase on #832

File: tests/hurl/next_stakes.hurl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    chainId
+    delegate
+    epoch
+    ledgerHash
+    nonce
+    pk
+    public_key
+    token
+    voting_for
+    delegationTotals {
```
Comment (jhult) [inline tests/hurl/next_stakes.hurl:15]:
```suggestion
    nextDelegationTotals {
```

File: tests/hurl/next_stakes.hurl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+jsonpath "$.data.nextStakes[0].chainId" == "5f704c"
+jsonpath "$.data.nextStakes[0].delegate" == "B62qiburnzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzmp7r7UN6X"
+jsonpath "$.data.nextStakes[0].epoch" == 42
+jsonpath "$.data.nextStakes[0].ledgerHash" == "jxYFH645cwMMMDmDe7KnvTuKJ5Ev8zZbWtA73fDFn7Jyh8p6SwH"
+jsonpath "$.data.nextStakes[0].nonce" == 0
+jsonpath "$.data.nextStakes[0].pk" == "B62qiburnzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzmp7r7UN6X"
+jsonpath "$.data.nextStakes[0].public_key" == "B62qiburnzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzzmp7r7UN6X"
+jsonpath "$.data.nextStakes[0].token" == 1
+jsonpath "$.data.nextStakes[0].voting_for" == "3NK2tkzqqK5spR2sZ7tujjqPksL45M3UUrcA4WhCkeiPtnugyE2x"
+jsonpath "$.data.nextStakes[0].delegationTotals.countDelegates" == 1
```
Comment (jhult) [inline tests/hurl/next_stakes.hurl:44]:
```suggestion
jsonpath "$.data.nextStakes[0].nextDelegationTotals.countDelegates" == 1
```

File: tests/hurl/next_stakes.hurl [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+jsonpath "$.data.nextStakes[0].delegationTotals.totalDelegated" == 2498323.273015384
```
Comment (jhult) [inline tests/hurl/next_stakes.hurl:45]:
```suggestion
jsonpath "$.data.nextStakes[0].nextDelegationTotals.totalDelegated" == 2498323.273015384
```

File: rust/src/web/graphql/next_stakes/mod.rs [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+}
+
+#[derive(SimpleObject)]
+pub struct NextStakesLedgerAccountWithMeta {
+    /// Value next epoch
+    epoch: u32,
+    /// Value next ledger hash
+    ledger_hash: String,
+    /// Value delegation totals
+    delegation_totals: NextStakesDelegationTotals,
```
Comment (jhult) [inline rust/src/web/graphql/next_stakes/mod.rs:149]:
```suggestion
    #[graphql(name = "nextDelegationTotals")]
    next_delegation_totals: NextStakesDelegationTotals,
```

File: rust/src/web/graphql/next_stakes/mod.rs [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+                    initial_minimum_balance: Some(timing.initial_minimum_balance),
+                    vesting_increment: Some(timing.vesting_increment),
+                    vesting_period: Some(timing.vesting_period),
+                });
+
+                NextStakesLedgerAccountWithMeta {
+                    epoch,
+                    ledger_hash: ledger_hash.clone(),
+                    account: NextStakesLedgerAccount::from(account),
+                    delegation_totals: NextStakesDelegationTotals {
```
Comment (jhult) [inline rust/src/web/graphql/next_stakes/mod.rs:117]:
```suggestion
                    next_delegation_totals: NextStakesDelegationTotals {
```

**Review by jhult — CHANGES_REQUESTED** [review]
To match [the current ME API](https://github.com/Granola-Team/mina-block-explorer/blob/main/graphql/queries/next_staking_ledgers.graphql#L15), we need to use `nextDelegationTotals` instead of `delegationTotals`.

File: rust/src/web/graphql/next_stakes/mod.rs [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+#[derive(InputObject)]
+pub struct NextStakesQueryInput {
+    epoch: Option<u32>,
+    delegate: Option<String>,
+    ledger_hash: Option<String>,
+
+    #[graphql(name = "public_key")]
+    public_key: Option<String>,
+}
```
Comment (jhult) [inline rust/src/web/graphql/next_stakes/mod.rs:19]: Should we just share `StakesQueryInput`?

```suggestion

```

File: rust/src/web/graphql/next_stakes/mod.rs [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+    token: u32,
+    /// Value receipt chain hash
+    #[graphql(name = "receipt_chain_hash")]
+    receipt_chain_hash: String,
+    /// Value voting for
+    #[graphql(name = "voting_for")]
+    voting_for: String,
+    /// Value balance nanomina
+    balance_nanomina: u64,
+}
```
Comment (jhult) [inline rust/src/web/graphql/next_stakes/mod.rs:182]: Should we just share `StakesLedgerAccount`?

```suggestion

```

File: rust/src/web/graphql/next_stakes/mod.rs [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+    pub cliff_amount: Option<u64>,
+    #[graphql(name = "cliff_time")]
+    pub cliff_time: Option<u64>,
+    #[graphql(name = "initial_minimum_balance")]
+    pub initial_minimum_balance: Option<u64>,
+    #[graphql(name = "vesting_increment")]
+    pub vesting_increment: Option<u64>,
+    #[graphql(name = "vesting_period")]
+    pub vesting_period: Option<u64>,
+}
```
Comment (jhult) [inline rust/src/web/graphql/next_stakes/mod.rs:206]: Should we just share `StakesTiming`?

```suggestion

```

File: rust/src/web/graphql/next_stakes/mod.rs [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+            delegate,
+            pk,
+            public_key,
+            token,
+            receipt_chain_hash,
+            voting_for,
+            balance_nanomina,
+        }
+    }
+}
```
Comment (jhult) [inline rust/src/web/graphql/next_stakes/mod.rs:241]: If we share `StakesLedgerAccount`, we can remove this.

```suggestion
```

File: rust/src/web/graphql/next_stakes/mod.rs (hunk 4 above)
Comment (trevorbernard) [reply rust/src/web/graphql/next_stakes/mod.rs:149]: The explicit naming shouldn't be necessary since snake_case gets automatically converted to camelCase

**Review by jhult — APPROVED** [review]
Let's not block on my feedback. That can be handled in a subsequent PR.

**Pull request merged**